dotnet/src/Connectors/Connectors.AI.OpenAI/AzureSdk/SKChatMessage.cs

```diff
@@ -33,9 +59,32 @@ public SKChatMessage(string role, string content)
     {
     }
 
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SKChatMessage"/> class.
+    /// </summary>
+    /// <param name="role">Role of the author of the message.</param>
+    /// <param name="content">Content of the message.</param>
+    /// <param name="arguments">Arguments of the message.</param>
+    public SKChatMessage(string role, string content, Dictionary<string, string> arguments)
+        : base(new AuthorRole(role), content, arguments)
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SKChatMessage"/> class.
+    /// </summary>
+    /// <param name="role">Role of the author of the message.</param>
+    /// <param name="content">Content of the message.</param>
+    /// <param name="functionCall">Function call definition.</param>
+    /// <param name="arguments">Arguments of the message.</param>
+    public SKChatMessage(string role, string content, FunctionCall? functionCall, Dictionary<string, string> arguments)
+        : base(new AuthorRole(role), content, arguments)
+    {
+        this.FunctionCall = functionCall;
+    }
+
     /// <summary>
     /// Exposes the underlying OpenAI SDK function call chat message representation
```

Comment:
Why is a property specific to OpenAI on SKChatMessage?

Reply:
Oh, nevermind, this is the Azure OpenAI connector's ChatMessageBase-derived type.  Seems like SKChatMessage is the wrong name for the type.